.github/actions/generate-evm-artifacts/action.yml

```diff
@@ -0,0 +1,19 @@
+name: Generate EVM artifacts
+inputs:
+  node_version:
+    description: "Node version to use"
+    required: true
+runs:
+  using: "composite"
+  steps:
+    - name: "Use Node ${{ inputs.node_version }}"
+      uses: actions/setup-node@v3
+      with:
+        node-version: "${{ inputs.node_version }}"
+        cache: yarn
+    - name: Install packages
+      shell: bash
+      run: yarn install --frozen-lockfile
```

Comment:
This still have below errors here when building optional `node-hid` and `cpu-features`. We cannot do `--ignore-optional` here like in other places as `hardhat compile` depends on other optional dependencies being installed.

```
warning Error running install script for optional dependency: "/home/runner/work/contracts/contracts/node_modules/node-hid: Command failed.
info This module is OPTIONAL, you can safely ignore this error

warning Error running install script for optional dependency: "/home/runner/work/contracts/contracts/node_modules/cpu-features: Command failed.
info This module is OPTIONAL, you can safely ignore this error
```